[bloomsky] BloomSky Binding initial contribution

dschoepel opened the pull request:
Signed-off-by: Dave J Schoepel <dave@theschoepels.com>

<!--
Thanks for contributing to the openHAB project!
Please describe the goal and effect of your PR here.
Pay attention to the below notes and to *the guidelines* for this repository.
Feel free to delete any comment sections in the template (starting with "<!--").
-->

<!-- TITLE -->
[bloomsky] Initial contribution
<!--
Please provide a PR summary in the *Title* above, according to the following schema:
- If related to one specific add-on: Mention the add-on shortname in square brackets
  e.g. "[exec]", "[netatmo]" or "[tesla]"
- If the PR is work in progress: Add "[WIP]"
- Give a short meaningful description in imperative mood
  e.g. "Add support for device XYZ" or "Fix wrongly handled exception"
  for a new add-on/binding: "Initial contribution"
Examples:
- "[homematic] Improve communication with weak signal devices"
- "[timemachine][WIP] Initial contribution"
- "Update contribution guidelines on new signing rules"
-->

<!-- DESCRIPTION -->
Implementing an OpenHab binding for the BloomSky personal weather station.  This is a "read-only" binding that uses the [BloomSky API](http://weatherlution.com/bloomsky-api/?doing_wp_cron=1615241711.4678061008453369140625) to retrieve the sensor data from the **SKY1, SKY2** and **STORM** personal weather stations.  If you are not familiar with these weather stations, you can find out more at the [BloomSky Home Page](https://www.bloomsky.com/).

This pull request includes the initial working binding that has been developed against the 3.2.0-SNAPSHOT.  The coding guidelines have been followed, all issues from the static code analysis have been addressed.    

Link to JAR file: https://nextcloud.theschoepels.com/s/RHq2fGeE49dQ3Zg

<!--
Please give a few sentences describing the overall goals of the pull request.
Give enough details to make the improvement and changes of the PR understandable
to both developers and tech-savy users.

Please keep the following in mind:
- What is the classification of the PR, e.g. Bugfix, Improvement, Novel Addition, ... ?
- Did you describe the PRs motivation and goal?
- Did you provide a link to any prior discussion, e.g. an issue or community forum thread?
- Did you describe new features for the end user?
- Did you describe any noteworthy changes in usage for the end user?
- Was the documentation updated accordingly, e.g. the add-on README?
- Does your contribution follow the coding guidelines:
  https://www.openhab.org/docs/developer/guidelines.html
- Did you check for any (relevant) issues from the static code analysis:
  https://www.openhab.org/docs/developer/bindings/#include-the-binding-in-the-build
- Did you sign-off your work:
  https://www.openhab.org/docs/developer/contributing.html#sign-your-work
-->

<!-- TESTING -->
This binding has been tested against the developer's BloomSky account that has two Sky's and one Storm weather station associated with it.  It is currently running in a OH 3.2.0 system in both Windows and RPI.  

<!--
Your Pull Request will automatically be built and available under the following folder:
https://openhab.jfrog.io/openhab/libs-pullrequest-local/org/openhab/

It is a good practice to add a URL to your built JAR in this Pull Request description,
so it is easier for the community to test your Add-on.
If your Pull Request contains a new binding, it will likely take some time
before it is reviewed and processed by maintainers.
That said, consider submitting your Add-on in the Eclipse IoT Marketplace
See this thread for more info:
https://community.openhab.org/t/24491

Don't forget to submit a thread about your Add-on in the openHAB community:
https://community.openhab.org/c/add-ons 
-->

dschoepel commented:
New to this process and the Jenkins build seems to have failed.  Checking the console log there seems to be a problem with my binding not passing a spotless check along with several other bindings that are not part of my request.  I ran the spotless check prior to commit and this pull request and had no errors.  I re-ran the check and still do not see any errors.  Am I missing something?  Any help /advice is appreciated....

dalgwen commented on bundles/pom.xml:
```diff
@@ -60,7 +60,8 @@
     <module>org.openhab.binding.bigassfan</module>
-    <module>org.openhab.binding.bluetooth</module>
+    <module>org.openhab.binding.bloomsky</module>
+	<module>org.openhab.binding.bluetooth</module>
```
It seems you used on the bluetooth line a "\t" instead of the usual four spaces, thus the spotless check failing.

dschoepel replied:
Thanks for the help  @dalgwen  - so simple...

dschoepel commented:
@Skinah , The code-analysis reports are clean/empty (no warnings), and both checks have passed. Let me know if there is anything else I need to do...

Skinah commented on bundles/org.openhab.binding.bloomsky/src/main/resources/OH-INF/thing/thing-types.xml:
```diff
@@ -0,0 +1,547 @@
+	<channel-type id="windChill">
+		<item-type>Number:Temperature</item-type>
+		<label>Wind Chill</label>
+		<description>Wind chill in °F or °C</description>
```
remove any referrence to C or F as it is automatic due to your using UOM.

dschoepel replied:
Removed temp indicator as suggested.

Skinah commented on bundles/org.openhab.binding.bloomsky/src/main/resources/OH-INF/thing/thing-types.xml:
```diff
@@ -0,0 +1,547 @@
+
+	<channel-type id="rainDaily">
+		<item-type>Number:Length</item-type>
+		<label>Precipitation Total Past 24h</label>
```
We prefer the label to be 2-3 words max. Suggest:
Todays Rain
Rain Today
Rain Daily

I tend to prefer keeping it the same as the channel ID as it makes it easier for the textual people. The label is used to auto create the name for the items when you create equipment from a thing.

dschoepel replied:
Went through entire list of labels and made changes as suggested.

dschoepel commented:
@Skinah , I could use some advice on how to correct the DCO check failure.  I am not clear on what steps I need to take to fix the following error?  Thanks in advance....

Commit sha: b1c5108, Author: lolodomo, Committer: Dave J Schoepel; Expected "lolodomo lg.hc@free.fr", but got "Laurent Garnier lg.hc@free.fr".

dschoepel commented:
I need to close this pull request and open a new one as this branch has been corrupted by my efforts to get it cleaned up after doing a rebase out of sequence (still a GIT rookie).  I will open a new one and reference this so we don't lose the conversation.

dschoepel commented:
Bloomsky seems to have gone out of business and abandoned its customers.  The API is no longer accessible.  Removed this pull request and moved on to a different weather station.  Tempest by Weatherflow.
